src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -88,45 +146,99 @@ public void initializeConverter() {
     public void disposeConverter() {
         clusterColorControl.removeAttributeListener(this);
         clusterLevelControl.removeAttributeListener(this);
+        clusterOnOff.removeAttributeListener(this);
     }
 
     @Override
     public void handleRefresh() {
-        clusterColorControl.getCurrentHue(0);
-        clusterColorControl.getCurrentSaturation(0);
+        if(supportsHue) {
+            clusterColorControl.getCurrentHue(0);
+            clusterColorControl.getCurrentSaturation(0);
+        }
+        else {
+            clusterColorControl.getCurrentX(0);
+            clusterColorControl.getCurrentY(0);
+        }
+
         clusterLevelControl.getCurrentLevel(0);
+        clusterOnOff.getOnOff(0);
     }
 
     @Override
     public Runnable handleCommand(final Command command) {
         return new Runnable() {
+            protected void changeOnOff(OnOffType onoff) throws InterruptedException, ExecutionException {
+                if(onoff == OnOffType.ON) clusterOnOff.onCommand();
+                else clusterOnOff.offCommand();
```

Comment:
The formatter needs to be run on this I think? I think the formatter would add {...} around each clause and use separate lines - please use the formatter if it's not used now.

Reply:
Not using eclipse, just plain commandline javac, jar, and vi as editor. Did not think I would need to install the full dev platform for this, but looks like I will have to :)

Reply:
Ouch - vi? Really ;)

I would really recommend a 'proper' IDE as it will help report various coding errors as well as things like formatting...